### PR TITLE
handle key not found error

### PR DESCRIPTION
**Pull request by deepak1725**
# Description

Fixes key not found issue while requesting stack report.

Fixes # (issue)
https://jira.coreos.com/browse/APPAI-1025

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Comment by deepak1725** [comment]
f8a gemini server deepak1725 fabric8 analytics devtools dev ext devshift net api v1 stacks report report daily

**Comment by arajkumar** [comment]
@deepak1725 , It would be great if you add some explanation regarding the root cause of the issue.

**Comment by deepak1725** [comment]
> @deepak1725 , It would be great if you add some explanation regarding the root cause of the issue.

@arajkumar There is nothing to be explained. **Not found Error** was not handled. So added some code to handle same.

**Comment by arajkumar** [comment]
@deepak1725, Under what situation this exception occurs? Due to bad API input? Also the swagger spec need to be updated to reflect the new error code(400)?

**Comment by deepak1725** [comment]
> @deepak1725, Under what situation this exception occurs? 
Its when User hits unknown URL.
So, our API server understands `foo.com/bar` 
So now a user comes and request `foo.com/car`, Server will return **Not Found Error**.

Same is the case here... From logs, User is requesting `/prod` hence server looks into s3 for relevant key/path for a file, hence raises not found error. 

> Also the swagger spec needs to be updated to reflect the new error code(400)? 
Yes. This is a regular case of NOT found requested Stack Reports.

**Comment by arajkumar** [comment]
@deepak1725 , in that case you should return `404` not `400`. Also the error must have meaningful message, not the info from exception.

**Comment by deepak1725** [comment]
> @deepak1725 , in that case you should return `404` not `400`. Also the error must have meaningful message, not the info from exception.

Please reach out to @miteshvp for more clarification on this.

**Comment by miteshvp** [comment]
@deepak1725 - +1 to returning 404 and a custom message if a key is not found, or something generic for other cases.

**Comment by deepak1725** [comment]
> @deepak1725 - +1 to returning 404 and a custom message if a key is not found, or something generic for other cases.

@miteshvp done

File: src/rest_api.py [hunk 1]
```diff
@@ -452,7 +453,13 @@ def get_stacks_report(report):
     A report matching with the filename retrieved using the /stacks-report/list/{frequency}
     will be returned.
     """
-    return flask.jsonify(_s3_helper.get_object_content(report))
+    try:
+        return flask.jsonify(_s3_helper.get_object_content(report)), 200
+    except ClientError as e:
+        return flask.jsonify({
```
Comment (miteshvp) [inline src/rest_api.py]: Custom error should also be fine. I am ok with this approach, since it is internal to platform.

**Pull request merged**